source/_UIAHandler.py

```diff
@@ -312,7 +312,13 @@ def _isUIAWindowHelper(self,hwnd):
 		if appModule and appModule.isBadUIAWindow(hwnd):
 			return False
 		# Ask the window if it supports UIA natively
-		return windll.UIAutomationCore.UiaHasServerSideProvider(hwnd)
+		res=windll.UIAutomationCore.UiaHasServerSideProvider(hwnd)
+		if res:
+			# the window does support UIA natively, but
+			if windowClass=="_WwG" and not (config.conf['UIA']['useInMSWordWhenAvailable'] or not appModule.helperLocalBindingHandle):
+				# Microsoft Word should not use UIA unless we can't inject or the user explicitly chose to use UIA with Microsoft word
+				return False
+		return res

```

Comment:
The name of this function suggests a boolean return, while this can be implicitly be used as a boolean. I would rather this be changed to `return bool(res)`